app/models/authority.rb

```diff
@@ -1,3 +1,21 @@
 class Authority < ApplicationRecord
   has_many :registers
+
+  scope :by_name, -> { order name: :asc }
+
+  scope :with_a_register, -> {
+    joins(:registers).merge(Register.in_beta).distinct.by_name
+  }
+
+  scope :collection, ->(id) {
+    Authority.find_by(government_organisation_key: id)
```

Comment:
We should use [find_by!](https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find_by-21) here as we want to 404 if not found rather than 500.

Reply:
Changes in both the authority and category models.